runtime/ms-rest/lib/serialization.js

```diff
@@ -507,13 +511,16 @@ function deserializeCompositeType(mapper, responseBody, objectName) {
         if (modelProps[key].serializedName !== '') propertyObjectName = objectName + '.' + modelProps[key].serializedName;
         let propertyMapper = modelProps[key];
         let serializedValue;
+        let responseValue = responseBody[key];
         //paging
-        if (Array.isArray(responseBody[key]) && modelProps[key].serializedName === '') {
-          propertyInstance = responseBody[key];
+        if (Array.isArray(responseValue) && modelProps[key].serializedName === '') {
+          propertyInstance = responseValue;
           instance = exports.deserialize.call(this, propertyMapper, propertyInstance, propertyObjectName);
         } else if ((propertyInstance !== null && propertyInstance !== undefined) || (propertyMapper && propertyMapper.isConstant)) {
           serializedValue = exports.deserialize.call(this, propertyMapper, propertyInstance, propertyObjectName);
           instance[key] = serializedValue;
+        } else if (!responseValue && propertyMapper.isPolymorphicDiscriminator){
```

Comment:
same as above. Let exports.deserialize.call() do its job. set responeBody[key] to mapper.serializedName around line 513.